.github/README.md

```diff
@@ -0,0 +1,24 @@
+# STLab CI Configuration
+
+The CI configuration in `workflows/build_and_test.yml` strives to be a minimal wrapper around the root-level `stlab.py` script, which will be invoked thrice (once each for `install`, `build`, and `test` subcommands) for each configuration listed in `.matrix.json`.
+
+Please avoid adding platform-specific build configuration directly in `build_and_test.yml` - this makes it difficult for developers to reliably reproduce our build configuration on a new machine. Instead, add the necessary platform-specifics to the root-level `stlab.py`, so the CI system, and our developers, can benefit from the change. 
+
+## Supported Versions
+
+Our CI system supports the matrix of operating systems, compilers, and compiler versions listed in `./matrix.json`. Other platforms may be supported but are not actively maintained.
```

Comment:
I think this section goes without saying. That `matrix.json` file is illegible to humans BTW. Does it need to be a single line?